Modules/booking/Controller/BookingdefaultController.php

```diff
@@ -1060,8 +1040,67 @@ public function deleteperiodAction($id_space, $id_period){
         $this->redirect("booking/" . $id_space);
     }
 
+    public function checkValidationAuthorisation($id_space, $id_calEntry, $dateResaStart) {
+        $lang = $this->getLanguage();
+        $modelCalEntry = new BkCalendarEntry();
+        $modelResource = new ResourceInfo();
+        $modelBkAccess = new BkAccess();
+        $entryInfo = $modelCalEntry->getEntry($id_space, $id_calEntry);
+        $resource = $modelResource->get($id_space, $entryInfo['resource_id']);
+        $bkAccess = $modelBkAccess->getAccessId($id_space, $resource['id']);
+
+        $temp = explode("-", $dateResaStart);
+        try {
+            $curentDateUnix = mktime(0, 0, 0, intval($temp[1]), intval($temp[2]), intval($temp[0]));
+        } catch (Exception $e) {
+            Configuration::getLogger()->debug('[booking] invalid input date', ['date' => $entryInfo['start_time']]);
+            $curentDateUnix = time();
+        }
+
+        $canValidateBooking = $this->hasAuthorization(
+            $resource['id_category'],
+            $bkAccess, $id_space,
+            $_SESSION['id_user'],
+            $curentDateUnix
+        );
+
+        $redir = $this->request->getParameterNoException('from');
+
+        $backTo = [];
+        $redirPage = '';
+        if ($redir) {
```

Comment:
should be in a try/catch in case from is not formatted as expected or do not have the 5 elts.... in case of catch, just ignore the from